### DISCUSSION TITLE
Shall we project the points on the sphere before inserting them?

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7478] [Q&A] [answered]
Hello,

In [this example](https://doc.cgal.org/latest/Triangulation_on_sphere_2/Triangulation_on_sphere_2_2triang_on_sphere_proj_8cpp-example.html), the points are projected on the sphere before inserting them in the Delaunay triangulation. Isn't it automatic when we use `Projection_on_sphere_traits_3`? I mean does the insertion performs the projection when we use this class?

**Comment by MaelRL** [comment]
The `Construct_point_on_sphere_2` of that traits does not project points on the sphere, but merely caches a scaling factor, see: https://github.com/CGAL/cgal/blob/master/Triangulation_on_sphere_2/include/CGAL/Projection_on_sphere_traits_3.h#L157.

The name is somewhat confusing, that's true.

**Reply by stla** [reply]
Aahh ok now I understand why I got this strange Delaunay ball with my implementation:
![strangeDelaunay](https://github.com/CGAL/cgal/assets/4466543/06ecf1c8-92f7-40a6-89c6-38cb6b8ca3ee)
The vertices have not been projected as I believed!

**Reply by MaelRL** [reply, marked as answer]
You can call `get_projection()` on the objects of type `Point_on_sphere_2`

**Reply by stla** [reply]
Are you sure? I don't find this function in the doc, nor with Google.

**Reply by MaelRL** [reply]
It's an internal function: https://github.com/CGAL/cgal/blob/master/Triangulation_on_sphere_2/include/CGAL/Projection_on_sphere_traits_3.h#L53

**Reply by stla** [reply]
Ok I get it:
```cpp
  ......
  Traits ball(SPoint3(O(0), O(1), O(2)), radius);
  // projection on this ball
  Traits::Construct_point_on_sphere_2 projection =
    ball.construct_point_on_sphere_2_object();
  // Rcpp matrix to store the projected vertices
  Rcpp::NumericMatrix Vertices(3, npoints);
  // make Delaunay triangulation
  DToS dtos(ball);
  {
    int i = 0;
    for(const auto& pt : points) {
      Traits::Point_on_sphere_2 pos = projection(pt);
      DToS::Vertex_handle vh = dtos.insert(pos);
      SPoint3 p = projection(pos).get_projection(ball.center(), ball.radius());
      ......
```
Thanks!

**Reply by stla** [reply]
ah wait, I don't need `projection` a second time, surely

**Reply by MaelRL** [reply]
The triangulation's point type is `Traits::Point_on_sphere_2`, so it should be fine with `dtos.point(vh).get_projection(ball.center(), ball.radius())`...

**Reply by stla** [reply]
It also works with `pos.get_projection(ball.center(), ball.radius())`.
